src/main/java/com/target/ready/library/system/Controller/LibraryController.java

```diff
@@ -18,10 +18,20 @@ public class LibraryController {
     LibraryController(LibrarySystemService librarySystemService){
         this.librarySystemService=librarySystemService;
     }
-    @GetMapping("getBooks")
-    public List<Book> getAllBooks(){
-
-        return librarySystemService.getAllBooks();
+    @GetMapping("books_directory")
+    public List<Book> getAllBooks(@RequestParam(value = "pageNumber", defaultValue = "0", required = false) Integer pageNumber){
+        List<Book> books;
+        int pageSize=5;
+        System.out.println("page num:"+pageNumber);
```

Comment:
Please use logger instead of using Sysout.

Reply:
okay will do it